### ISSUE TITLE
Add request handlers to v2 implementation

### ISSUE THREAD

**Issue by randomvariable** [kubernetes/cloud-provider-aws#137]
**What would you like to be added**:
Follow up from #131, add AWS request handlers to do the following:

* Add metrics
* Add rate limiting

@M00nF1sh , do you think it's worth re-implementing what's done in inhttps://github.com/aws/aws-app-mesh-controller-for-k8s/tree/master/pkg/aws for the above?

FWIW, I used the same metric formats in https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/master/pkg/cloud/metrics/metrics.go, so would be worth standardising.

**Why is this needed**:

/kind feature

**Comment by M00nF1sh** [comment]
@randomvariable Thanks for reaching out. I think we can keep same format(i'm open to new metrics if applicable) :D

**Comment by randomvariable** [comment]
@M00nF1sh Are we ok to copy the throttle package over to this repo and cluster-api?

**Comment by M00nF1sh** [comment]
@randomvariable of course. but i do think the throttle package needs some enhancements.
e.g. it should use endpointID instead of serviceName when throttling. since some service's serviceName have space within it.

**Comment by nckturner** [comment]
I spoke with @jaypipes and he is going to create a repository under the aws organization to put the throttle package and anything else.  @M00nF1sh WDYT about that?

**Comment by randomvariable** [comment]
@jaypipes has any progress been made on creating the repo to include importable limiters?
